File: frontend/src/types/rescue.ts
// 救助类型定义

export type AnimalType = 'cat' | 'dog' | 'other';
export type HealthStatus = 'healthy' | 'injured' | 'sick';
export type RescueStatus = 'pending' | 'rescuing' | 'rescued' | 'medical' | 'adopted' | 'closed';
export type ReviewStatus = 'pending' | 'approved' | 'rejected';

export interface RescueLocation {
  lat: number;
  lng: number;
  address?: string;
}

export interface Rescue {
  id: number;
  userId: number;
  animalType: AnimalType;
  healthStatus: HealthStatus;
  rescueStatus: RescueStatus;
  locationLat: number;
  locationLng: number;
  address?: string;
  description?: string;
  breed?: string;
  age?: string;
  contactPhone: string;
  contactWechat?: string;
  images: string[];
  reviewStatus: ReviewStatus;
  reviewComment?: string;
  createTime: string;
  updateTime: string;
}

export interface RescuePublishRequest {
  animalType: AnimalType;
  healthStatus: HealthStatus;
  locationLat: number;
  locationLng: number;
  address?: string;
  description?: string;
  breed?: string;
  age?: string;
  contactPhone: string;
  contactWechat?: string;
  images: string[];
}

export interface RescueDetailResponse extends Rescue {
  nickname?: string;
  avatar?: string;
}

export interface StatusUpdateRequest {
  status: RescueStatus;
  progress?: string;
  images?: string[];
}

// ===== 领养类型 =====

export type Age = 'young' | 'adult' | 'senior';
export type Gender = 'male' | 'female';
export type Size = 'small' | 'medium' | 'large';
export type AdoptionStatus = 'available' | 'pending' | 'adopted' | 'closed';

export interface HealthStatus {
  vaccinated?: boolean;
  dewormed?: boolean;
  neutered?: boolean;
}

export interface Adoption {
  id: number;
  userId: number;
  name?: string;
  animalType?: AnimalType;
  breed: string;
  age: Age;
  gender: Gender;
  size?: Size;
  personality: string;
  healthStatus: HealthStatus;
  requirements: string;
  story?: string;
  images: string[];
  contactPhone?: string;
  contactWechat?: string;
  adoptionStatus: AdoptionStatus;
  reviewStatus: ReviewStatus;
  reviewComment?: string;
  createTime: string;
  updateTime: string;
}

export interface AdoptionPublishRequest {
  name?: string;
  animalType?: AnimalType;
  breed: string;
  age: Age;
  gender: Gender;
  size?: Size;
  personality: string;
  healthStatus: HealthStatus;
  requirements: string;
  story?: string;
  images: string[];
  contactPhone?: string;
  contactWechat?: string;
}

export interface AdoptionDetailResponse extends Adoption {
  nickname?: string;
  avatar?: string;
}

// ===== 地图类型 =====

export type MarkerColor = 'red' | 'yellow' | 'green' | 'blue' | 'gray';

export interface MapPoint {
  id: number;
  type: 'rescue' | 'adoption';
  markerColor: MarkerColor;
  lat: number;
  lng: number;
  title: string;
  status: string;
  animalType?: AnimalType;
  thumbnailImages?: string[];
  createTime: string;
}

export interface HeatmapPoint {
  lat: number;
  lng: number;
  intensity: number; // 1-10
}

export interface MapPointsParams {
  lat: number;
  lng: number;
  radius?: number;
  types?: 'rescue' | 'adoption' | 'rescue,adoption';
}

export interface HeatmapParams {
  lat: number;
  lng: number;
  radius?: number;
}

File: backend/feed-service/src/main/java/com/petpal/feed/model/Rescue.java
package com.petpal.feed.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.Instant;

@Data
@TableName("t_rescue")
public class Rescue implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    private Long userId;

    /**
     * 动物类型: cat, dog, other
     */
    private String animalType;

    /**
     * 健康状态: healthy, injured, sick
     */
    private String healthStatus;

    /**
     * 救助状态: pending, rescuing, rescued, medical, adopted, closed
     */
    private String rescueStatus;

    /**
     * 位置纬度
     */
    private BigDecimal locationLat;

    /**
     * 位置经度
     */
    private BigDecimal locationLng;

    /**
     * 地址描述
     */
    private String address;

    /**
     * 描述
     */
    private String description;

    /**
     * 品种
     */
    private String breed;

    /**
     * 年龄
     */
    private String age;

    /**
     * 联系电话
     */
    private String contactPhone;

    /**
     * 微信联系方式
     */
    private String contactWechat;

    /**
     * 图片URL数组 (JSON格式)
     */
    private String images;

    /**
     * 审核状态: pending, approved, rejected
     */
    private String reviewStatus;

    /**
     * 审核备注
     */
    private String reviewComment;

    @TableField(fill = FieldFill.INSERT)
    private Instant createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Instant updateTime;

    @TableLogic
    private Integer deleted;
}
